packages/ibm-products/src/components/Coachmark/next/Coachmark/CoachmarkBeacon/CoachmarkBeacon.test.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright IBM Corp. 2024, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react'; // https://testing-library.com/docs/react-testing-library/intro
+
+import { pkg } from '../../../../../settings';
+import uuidv4 from '../../../../../global/js/utils/uuidv4';
+import { Coachmark } from '../Coachmark';
+import { CoachmarkBeacon } from '.';
+import { Button } from '@carbon/react';
+
+const blockClass = `${pkg.prefix}--coachmark-beacon`;
+const componentName = CoachmarkBeacon.displayName;
+
+// values to use
+const childDataTestId = `beacon-${uuidv4()}`;
+const className = `class-${uuidv4()}`;
+
+const renderCoachmarkWithBeacon = ({ ...rest } = {}) =>
+  render(
+    <Coachmark>
+      <CoachmarkBeacon
+        id="CoachmarkBtn"
+        label="Show information"
+        data-testid={childDataTestId}
+        {...rest}
+      ></CoachmarkBeacon>
+      <Coachmark.Content highContrast={true}>
+        <Coachmark.Content.Header></Coachmark.Content.Header>
+        <Coachmark.Content.Body>
+          <h2>Hello World</h2>
+          <p>this is a description test</p>
+          <Button size="sm">Done</Button>
+        </Coachmark.Content.Body>
+      </Coachmark.Content>
+    </Coachmark>
+  );
+
+describe(componentName, () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'warn').mockImplementation(() => {});
+  });
+  it('renders a component CoachmarkBeacon', () => {
+    renderCoachmarkWithBeacon({ label: 'Show information' });
+    expect(screen.getByTestId(childDataTestId)).toHaveClass(blockClass);
+  });
+
+  it('has no accessibility violations', async () => {
+    const { container } = renderCoachmarkWithBeacon({
+      label: 'Show information',
+    });
+    await expect(container).toBeAccessible(componentName);
+    await expect(container).toHaveNoAxeViolations();
+  });
+
+  it('applies className to the containing node', () => {
+    renderCoachmarkWithBeacon({ className, label: 'Show information' });
+    expect(screen.getByTestId(childDataTestId)).toHaveClass(className);
+  });
+
+  it('adds additional props to the containing node', () => {
+    const testingLabel = `testing-labels-${uuidv4()}`;
+    renderCoachmarkWithBeacon({
+      label: testingLabel,
+    });
+    screen.getAllByLabelText(testingLabel)[1];
```

Comment:
I think use `getByLabelText` instead of `getAllByLabelText` might be better here since you're looking for a specific label